src/DDTrace/OpenTracer/ScopeManager.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace DDTrace\OpenTracer;
+
+use DDTrace\Contracts\ScopeManager as ScopeManagerInterface;
+use OpenTracing\ScopeManager as OTScopeManager;
+use OpenTracing\Span as OTSpan;
+
+final class ScopeManager implements OTScopeManager
+{
+    /**
+     * @var ScopeManagerInterface
+     */
+    private $scopeManager;
+
+    /**
+     * @param ScopeManagerInterface $scopeManager
+     */
+    public function __construct(ScopeManagerInterface $scopeManager)
+    {
+        $this->scopeManager = $scopeManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function activate(OTSpan $span, $finishSpanOnClose = true)
+    {
+        return $this->scopeManager->activate(
+            Span::toDDSpan($span),
```

Comment:
I may need some guidance here because I am not sure this would work! Typically this method will not be called by the user, it will be called by our `DDTrace\OpenTracer\Tracer::startActiveSpan()`. If this is true then we don't need to "copy" the Span as it is risky and will not be kept in sync (also looking at the actual implementation looks like we do not transfer tags?), because in that case the OTSpan that we receive would already be our wrapped span, so not need to copy into our own.

There is a corner case here, when users provide as with their own `OpenTracing\Span` implementation, which we should honor. I think that in that case what we should do is to wrap it into another wrapper which implements our interface, but still use it as storage, as this way we don't have to keep it in sync.

WDYT?

Reply:
You're totally right - this won't work. Lol. Nice catch! There's another place or two where I do this with other OT objects. This was part of one of my failed attempts to wrap everything. I'm going to change this to check if the object is an instance of our OT implementation like I did [in the tracer](https://github.com/DataDog/dd-trace-php/pull/193/files#diff-e20592803012a46049e8d5df3097b7f7R65) and if so we can unwrap it. Otherwise, we'll fall back to another wrapper like you suggested. :)